src/smt_util/node_traversal.h

```diff
@@ -0,0 +1,117 @@
+/*********************                                                        */
+/*! \file node_traversal.h
```

Comment:
I think that the appropriate location for this file is in `src/expr` (similar to `node_algorithm`) because it is not really connected to SMT solving per se. Opinions @mpreiner @aniemetz?

Reply:
Yes, should be the same as node_algorithm.

Reply:
Done in 19a0080ef

Reply:
We'll probably want to move [the node visitor](https://github.com/CVC4/CVC4/blob/master/src/smt_util/node_visitor.h) at some point too.